drafting/conda-template.py

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python
+
+# This script reads in a yaml file (qiime2 recipe)
+# Pulls the needed data from that file
+# And uses that data to write to a secondary yaml file
+# Which will be the end result: a conda recipe
+
+import yaml
+import os
+import sys
+from jinja2 import Template
+
+if __name__ == '__main__':
+    # Reading the qiime2 recipe file
+    try:
+        qiime2_recipe = sys.argv[1]
+    except:
+        raise ValueError('Missing required parameter. QIIME 2 recipe filepath must be provided.')
+    
+    with open(qiime2_recipe) as qiime2_recipe:
+        parsed_recipe = yaml.load(qiime2_recipe, Loader=yaml.FullLoader)
+
+    try:
+        filepath = sys.argv[2]
+    except:
+        raise ValueError('Missing required parameter. conda recipe filename must be provided.')
+    
+    if os.path.exists(filepath):
+        raise FileExistsError('Invalid filename. Recipe file already exists.')
+
+    # Accessing jinja template with recipe key pairs
+    with open('jinja-template.j2') as file:
+        template = Template(file.read())
+
+    recipe_reqs = template.render(parsed_recipe)
+    # print(recipe_reqs)
+    conda_recipe = open(filepath, 'w')
+    yaml.dump(recipe_reqs, conda_recipe)
+    conda_recipe.close
```

Comment:
This should be another [context manager](https://www.python.org/dev/peps/pep-0343/), like above. Please take a read through of the pep linked to here, let's discuss some time!

Reply:
The rendered file looks a little funky, probably because you have an extra `yaml.dump` here. I don't think you need that in conjunction with your jinja template.

```
"package:\n    name: q2-cutadapt\n    version: 1\nsource:\n    path: .\nbuild:\n \
  \   script: make install\nrequirements:\n    host:\n        - python \n        -\
  \ setuptools \n    run:\n        - python \n        ['qiime2', 'q2-types']\n   \
  \     ['cutadapt >=3', 'pandas {{ pandas }}', 'numpy {{ numpy }}', 'pigz']\ntest:\n\
  \    imports: ['q2_cutadapt', 'from qiime2.plugins import cutadapt']\n    commands:\
  \ ['pytest -k q2_cutadapt']"
```

Reply:
@thermokarst ah, okay yeah I was wondering about that - that's helpful, thank you!!